src/main/java/org/opentripplanner/routing/core/State.java

```diff
@@ -48,25 +49,12 @@ public class State implements Cloneable {
 
   /* CONSTRUCTORS */
 
-  /**
-   * Create an initial state, forcing vertex to the specified value. Useful for tests, etc.
-   */
-  public State(Vertex vertex, RouteRequest opt, StreetMode streetMode) {
-    // Since you explicitly specify, the vertex, we don't set the backEdge.
-    this(vertex, opt.dateTime(), StateData.getInitialStateData(opt, streetMode));
-  }
-
-  /**
-   * Create an initial state, forcing vertex, back edge, time and start time to the specified
-   * values. Useful for starting a multiple initial state search, for example when propagating
-   * profile results to the street network in RoundBasedProfileRouter.
-   */
-  public State(Vertex vertex, Instant startTime, StateData stateData) {
+  public State(Vertex vertex, Instant startTime, StateData stateData, AStarRequest request) {
```

Comment:
Since we use the static "constructor" can this be private?

Reply:
We do not use it everywhere, there are eg. tests using this